### PR TITLE
Update operators with new tasks API changes

### PR DESCRIPTION
**Pull request by zen-dog**

### CODE REVIEW

**Comment by zen-dog** [comment]
This PR is against the `dev` branch, which is the new "bleeding edge" used to [test KUDO master](https://github.com/kudobuilder/kudo/pull/981/). And, of course, the ci tests are failing, since there is no released `kubectl-kudo` version, compatible with these API changes :( It's a chicken-and-egg problem.

File: repository/zookeeper/operator/operator.yaml [hunk 1]
```diff
@@ -40,14 +51,15 @@ plans:
...
     strategy: serial
     phases:
       - name: connection
-        strategy: parallel
+        strategy: serial
         steps:
           - name: connection
             tasks:
               - validation
-            delete: true
```
Comment (zen-dog) [inline repository/zookeeper/operator/operator.yaml:53]: This is fun: this step was removing a non-existing Job and so was always succeeding ¯\_(ツ)_/¯

**Review by alenkacz — APPROVED** [review]
very nice!

**Comment by zen-dog** [comment]
This PR (and subsequently `dev` branch) won't be green since tests are run with a `0.7.2` kudo version. We need to publish KUDO's `latest` binaries somewhere and use them here.

**Pull request merged**